File: VaultGuard.Android/app/src/main/java/com/vaultguard/app/ui/home/VaultSection.kt
package com.vaultguard.app.ui.home

import com.vaultguard.app.data.model.ItemType
import com.vaultguard.app.data.model.VaultItem

/** The selectable sidebar sections, mirroring the WPF navigation exactly. */
sealed class VaultSection(val title: String) {
    data object AllItems : VaultSection("All Items")
    data object Favorites : VaultSection("Favourites")
    data object Archive : VaultSection("Archive")
    data object RecentlyDeleted : VaultSection("Recently Deleted")
    data class Category(val type: ItemType, val label: String) : VaultSection(label)

    fun matches(item: VaultItem): Boolean = when (this) {
        AllItems -> !item.isArchived && !item.isDeleted
        Favorites -> item.isFavorite && !item.isArchived && !item.isDeleted
        Archive -> item.isArchived && !item.isDeleted
        RecentlyDeleted -> item.isDeleted
        is Category -> item.type == type && !item.isArchived && !item.isDeleted
    }

    companion object {
        // The five category filters shown in the WPF "Categories" group.
        val categories = listOf(
            Category(ItemType.Login, "Logins"),
            Category(ItemType.CreditCard, "Credit Cards"),
            Category(ItemType.SecureNote, "Secure Notes"),
            Category(ItemType.WiFi, "Wi-Fi"),
            Category(ItemType.Identity, "Identities"),
        )
    }
}

File: VaultGuard.iOS.Swift/VaultGuard/Features/Home/VaultSection.swift
// VaultSection.swift — the sidebar sections, mirroring the WPF navigation.
import Foundation

enum VaultSection: Hashable, Identifiable {
    case allItems, favorites, archive, recentlyDeleted
    case category(ItemType)

    var id: String { title }

    var title: String {
        switch self {
        case .allItems: return "All Items"
        case .favorites: return "Favourites"
        case .archive: return "Archive"
        case .recentlyDeleted: return "Recently Deleted"
        case .category(let t):
            switch t {
            case .login: return "Logins"
            case .creditCard: return "Credit Cards"
            case .secureNote: return "Secure Notes"
            case .wifi: return "Wi-Fi"
            case .identity: return "Identities"
            case .passkey: return "Passkeys"
            default: return t.label
            }
        }
    }

    var systemImage: String {
        switch self {
        case .allItems: return "square.stack.3d.up"
        case .favorites: return "star"
        case .archive: return "archivebox"
        case .recentlyDeleted: return "trash"
        case .category(let t): return t.systemImage
        }
    }

    func matches(_ item: VaultItem) -> Bool {
        switch self {
        case .allItems: return !item.isArchived && !item.isDeleted
        case .favorites: return item.isFavorite && !item.isArchived && !item.isDeleted
        case .archive: return item.isArchived && !item.isDeleted
        case .recentlyDeleted: return item.isDeleted
        case .category(let t): return item.type == t && !item.isArchived && !item.isDeleted
        }
    }

    static let categories: [VaultSection] = [
        .category(.login), .category(.creditCard), .category(.secureNote),
        .category(.wifi), .category(.identity), .category(.passkey),
    ]
}
